CPT331.iOS/CPT331.iOS/CDSuburb.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Suburb.swift
+//  SpatioNews
+//
+//  Created by Peter Weller on 16/09/2016.
+//  Copyright © 2016 Peter Weller. All rights reserved.
+//
+
+import UIKit
+import CoreLocation
+import CoreData
+
+class CDSuburb: NSManagedObject {
+    @NSManaged var name: String
+    @NSManaged var state: CDState
+    @NSManaged var latitude: Double
+    @NSManaged var longitude: Double
+    @NSManaged var postcode: Int
+    
+    lazy var location:CLLocation = CLLocation(latitude: self.latitude, longitude: self.longitude)
+    
+    class func create(name:String, state:CDState, latitude:Double, longitude:Double, postcode:Int?=nil) -> CDSuburb? {
+        let suburb = self.initEntity(withType: "Suburb") as! CDSuburb
+        
+        suburb.name = name
+        suburb.state = state
+        suburb.latitude = latitude
+        suburb.longitude = longitude
+        
+        if postcode != nil {
+            suburb.postcode = postcode!
+        }
+        
+        if suburb.save() {
+            return suburb
+        } else {
+            return nil
+        }
+    }
+    
+    
+    // Use haversine formula to calculate distances
```

Comment:
Interesting. I would have just used d = SQRT((X2 - X1)^2 + (Y2 - Y1)^2) - that being said it looks like Haversine could be potentially more accurate, especially over larger distances.